code/modules/mob/living/basic/vermin/cockroach.dm

```diff
@@ -0,0 +1,152 @@
+/mob/living/basic/cockroach
+	name = "cockroach"
+	desc = "This station is just crawling with bugs."
+	icon_state = "cockroach"
+	icon_dead = "cockroach" //Make this work
+	density = FALSE
+	mob_biotypes = MOB_ORGANIC|MOB_BUG
+	mob_size = MOB_SIZE_TINY
+	health = 1
+	maxHealth = 1
+	speed = 1.25
+	gold_core_spawnable = FRIENDLY_SPAWN
+	pass_flags = PASSTABLE | PASSGRILLE | PASSMOB
+
+	verb_say = "chitters"
+	verb_ask = "chitters inquisitively"
+	verb_exclaim = "chitters loudly"
+	verb_yell = "chitters loudly"
+	response_disarm_continuous = "shoos"
+	response_disarm_simple = "shoo"
+	response_harm_continuous = "splats"
+	response_harm_simple = "splat"
+	speak_emote = list("chitters")
+
+	//loot = list(/obj/effect/decal/cleanable/insectguts) Armhulen is making an element for this
+
+	basic_mob_flags = DEL_ON_DEATH
+	faction = list("hostile")
+
+	ai_controller = /datum/ai_controller/basic_controller/cockroach
+
+/mob/living/basic/cockroach/Initialize()
+	. = ..()
+	AddElement(/datum/element/swabable, CELL_LINE_TABLE_COCKROACH, CELL_VIRUS_TABLE_GENERIC_MOB, 1, 7)
+	AddElement(/datum/element/basic_body_temp_sensetive, 270, INFINITY)
+	AddComponent(/datum/component/squashable, squash_chance = 50, squash_damage = 1)
+	ADD_TRAIT(src, TRAIT_VENTCRAWLER_ALWAYS, INNATE_TRAIT)
+
+/mob/living/basic/cockroach/death(gibbed)
+	if(GLOB.station_was_nuked) //If the nuke is going off, then cockroaches are invincible. Keeps the nuke from killing them, cause cockroaches are immune to nukes.
+		return
+	..()
+
+/mob/living/basic/cockroach/ex_act() //Explosions are a terrible way to handle a cockroach.
+	return FALSE
+
+
+	/* this needs to become a subtree for cockroaches
+	if(turns_since_scan > time_to_hunt)
+		turns_since_scan = 0
+		var/list/target_types = list(/obj/effect/decal/cleanable/ants)
+		for(var/obj/effect/decal/cleanable/ants/potential_target in view(2, get_turf(src)))
+			if(potential_target.type in target_types)
+				hunt(potential_target)
+				return*/
+
+/datum/ai_controller/basic_controller/cockroach
+	blackboard = list(BB_TARGETTING_DATUM = new /datum/targetting_datum/basic())
+	ai_traits = STOP_MOVING_WHEN_PULLED
+	ai_movement = /datum/ai_movement/basic_avoidance
+	planning_subtrees = list(/datum/ai_planning_subtree/random_speech/cockroach)
+
+
+/datum/ai_controller/basic_controller/cockroach/PerformIdleBehavior(delta_time)
+	. = ..()
+	var/mob/living/living_pawn = pawn
+
+	if(DT_PROB(25, delta_time) && (living_pawn.mobility_flags & MOBILITY_MOVE) && isturf(living_pawn.loc) && !living_pawn.pulledby)
+		var/move_dir = pick(GLOB.alldirs)
+		living_pawn.Move(get_step(living_pawn, move_dir), move_dir)
+
+
+/obj/projectile/glockroachbullet
```

Comment:
this should be a bullet subtype

Reply:
it probably should but this code is taken verbatim from another file so its not really my problem